### ISSUE TITLE
Add skeleton loading state for UPE payment methods in blocks

### ISSUE THREAD

**Issue by FangedParakeet** [Automattic/woocommerce-payments#5397]
### Description

This is mainly noticeable when the split-UPE feature is enabled. Please see pdpOw2-Pn-p2#comment-2162 and surrounding conversation for more context.

Currently, in the blocks checkout, UPE elements are only mounted when their parent gateway is selected at checkout. This means the initial payment intent is created and the resulting element mounted only after the gateway radio button is selected. While these processes are occurring, the gateway at checkout remains static, jolting into life only once the element is finally ready to mount. Please lower your gaze to the below animation to see what I'm talking about.

![Janky UPE blocks loading](https://cdn-std.droplr.net/files/acc_1104206/ktwVqJ)
_This is what I'm talking about_

This issue is to add a skeleton loading state to this section while the element is being prepared. On the classic elements, there is [a ready state](https://stripe.com/docs/js/element/events/on_ready); if this is available for the Stripe React component as well, this would be an appropriate event to use to hide the loading state and show the mounted payment element.

### Acceptance criteria

* Add skeleton loading state for UPE payment methods in blocks checkout.
* Loading state removed when payment element is ready and mounted.

**Comment by zmaglica** [comment]
I noticed that this issue is in Heisenberg backlog. @frosso does the team have this issue under its radar and do you folks plan to work on this issue? The reason why I am asking is that I have to prioritize it, and it looks like to me that this issue falls under your scope, so you can maybe prioritize it on your end.

**Comment by FangedParakeet** [comment]
Just FYI, this is definitely under our radar and I have sort of begun resolving it already in a separate issue, by implementing a `LoadableBlock` element around the blocks payment elements, over [here](https://github.com/Automattic/woocommerce-payments/pull/5496/commits/ab81e884992cb41f5cee2307bbe82e11e5e19b89).

I think it would be nice to get this into our upcoming release, but we have a few other required issues that are ahead in terms of priority and since this is a non-essential enhancement, I'm not eager to commit wholeheartedly to completing this task with complete certainty within this sprint, since we will have limited availability due to the meetup next week.

**Comment by timur27** [comment]
This is the same question I’ve asked [here](https://app.zenhub.com/workspaces/heisenberg-63586f2ec60359001310fa6a/issues/gh/automattic/woocommerce-payments/5502).


@FangedParakeet If we assume the enhanced UPE is implemented, would this issue still be relevant? I understand it the way that the issue won’t be relevant since we will not have split UPE anymore. Am I missing something here?

If the above is confirmed and the issue will be irrelevant after the enhanced UPE is completed, then the question is if we still want to direct our efforts toward improving these design misalignments in order for them to be ideal before the enhanced UPE rollout?

**Comment by bborman22** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Heisenberg-63586f2ec60359001310fa6a/issues/Automattic/woocommerce-payments/5397?planning-poker) with Zenhub @FangedParakeet @timur27 @mdmoore

**Comment by FangedParakeet** [comment]
With our new deferred intent UPE implementation and with gratitude to Stripe, we now have an automatically enabled preloading state for these payment elements at checkout. 

![Preloading state for payment elements in blocks checkout](https://cdn-std.droplr.net/files/acc_1104206/v0LPSc)
_Preloading party_

This preloading state is present on both blocks and shortcode checkouts. I'm closing this as completed, but if anyone feels that this implementation is still incomplete, please feel free to reopen and preferably provide a more desired design specification instead.
